### DISCUSSION TITLE
Trying to play with sonicjs and following the directions in the readme, I get the error...

### DISCUSSION THREAD

**Discussion by pblanton** [SonicJs-Org/sonicjs#63] [Q&A] [answered]
"Error: ENOENT: no such file or directory, open '/home/pedro/Projects/sonicjs/server/themes/front-end/undefined/css/template.css'"

Apparently the environment variables aren't set and there is no mention of setting them in the root's readme.md file or in  your documentation (https://sonicjs.com/docs#prerequisites).

I'll probably get it figured out before you answer this, but it'd be nice for the next guy if you could document the actual steps to get the app running, because ...

```
$ git clone https://github.com/lane711/sonicjs
$ cd sonicjs
$ npm install
$ npm start
```

Is clearly not enough.

**Comment by pblanton** [comment]
I am following your recommendation here: https://sonicjs.com/docs#local-setup that says, 

"If this is your first time working with SonicJs, it is highly recommended that you use the out of the box file-based database."

And I am also seeing that that recommendation is in direct opposition to the instructions in the readme.md file at the repo root...

"Pre-requisites
You will need NodeJs installed and access to a MongoDB database. Its recommended that you setup MongoDB locally as it will run very fast, but you can also sign up for a free MongoDB database at Mlab: https://mlab.com/

Setup Steps
Assuming that you have a MongoDB database ready to go, setup takes < 2 minutes."

It seems that there are some huge gaps in your documentation. Maybe you are just expecting people to already know a bunch of stuff that you know??

**Comment by pblanton** [comment]
Ok. I found the code that copies the .env-default file to .env which "fixes" the original undefined theme error by setting the theme to "bootstrap". This is also where the MONGODB_URL comes from, but I am not using mongodb or wanting to use mongodb. I liked the idea that *"it is highly recommended that you use the out of the box file-based database."*

Now I am trying to figure out how to switch to a file-based back-end for my testing and setup locally.

This is important because I am also wanting to deploy it with a MySQL backend

**Comment by pblanton** [comment]
Ok. I found the install scripts in server/install/ and edited package.json to enable the npm-setup.js script...

```
"scripts": {
    "start": "node server.js",
    "dev": "nodemon server.js -e hbs,js,css,html --ignore 'template-processed.css' --ignore '*_spec.js'",
    "test": "npx cypress open --project ./tests",
    "setup": "node server/install/npm-setup.js"
  }, 
```

Then ran 'npm run setup' in order to initialize my mongo settings. Now I can run the app.

Now I need to figure out how to switch from Mongo to a flat-file. Not to be surly or anything, but I really wish this was in the documentation!

**Comment by lane711** [comment, marked as answer]
Hi, sorry for all the issues that you have experienced. The website docs are out of date. Please refer to the readme for the time being until I have time to update the site. The app now uses mongo db only as a back end

**Reply by pblanton** [reply]
When you said, "The app no uses mongo db only as a back end" did you mean that it "now" only uses Mongo?

If so then that's disappointing. I like Mongo OK, but I am looking for something that I can use in places where Mongo is not the easiest solution. Places like GoDaddy shared hosting where it's easy to use a json file-based backend, or MySQL.

The web is SLAM FULL of node.js solutions with a mongo back-end but that's not always the best option.

While I have your attention, do you have a deployment script that I haven't found yet? I'd like to spin up a deployment package and not have to manually copy-paste files. Anything you can do to point me in the right direction would be helpful.

**Reply by lane711** [reply]
Thanks for your feedback and I totally understand your thoughts. I suspect we'll add support for a flat file at some point in the future to support use cases like yours, but its not likely a near term goal. I had a lot of issues with Loopback ORM which is the tech that supported multiple databases and also wanted to move the API to graphql with wasn't supported with Loopback. 

For deployment, you can get a free mongo db from https://mlab.com/ and host a site for free with heroku - just make sure that the db and web server are in the same region. I'll have instructions for this soon once I am able to redo sonicjs.com with the latest version of the platform, but it should be pretty straightforward since the sonicjs repo already has a Procfile which is what heroku uses to launch your web app. - hth

**Comment by lane711** [comment]
Good news! SonicJs now uses TypeORM for it's data access layer. As a result, a flat file implementation is now possible. The default database is SQLite, which is a simple file based database. This should make it much easier to get SonicJs up and running locally without having to deal with a database.
